service/src/test/java/uk/gov/hmcts/reform/fpl/service/LocalAuthorityUserServiceTest.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.client.CaseAccessApi;
+import uk.gov.hmcts.reform.ccd.client.model.UserId;
+import uk.gov.hmcts.reform.fpl.config.LocalAuthorityUserLookupConfiguration;
+
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(SpringExtension.class)
+class LocalAuthorityUserServiceTest {
+
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String SERVICE_AUTH_TOKEN = "Bearer service token";
+    private static final String JURISDICTION = "PUBLICLAW";
+    private static final String CASE_TYPE = "Shared_Storage_DRAFTType";
+    private static final String USER_ID = "1";
+    private static final String CASE_ID = "1";
+    private static final String USER_TO_ADD = "5";
+    private static final String LOCAL_AUTHORITY = "example";
+
+    @Mock
+    private AuthTokenGenerator authTokenGenerator;
+    @Mock
+    private CaseAccessApi caseAccessApi;
+    @Mock
+    private LocalAuthorityUserLookupConfiguration localAuthorityUserLookupConfiguration;
+
+    @InjectMocks
+    private LocalAuthorityUserService localAuthorityUserService;
+
+    @Test
+    @SuppressWarnings({"LineLength"})
+    void shouldCallGrantAccessToCaseWithExpectedParams() {
```

Comment:
We should also unit test:
- make sure failure in granting access to first user does not stop other calls
- make sure exception is thrown when LA is unknown
- make sure exception is thrown when users list assigned to LA is blank (potentially)
- we could also make sure user crating case does not trigger call for itself as there is no need for that